### ISSUE TITLE
Full Labels in LaserMix

### ISSUE THREAD

**Issue by matildecc** [ldkong1205/LaserMix#34]
In your paper, you present a comparison with Full Labels (Fig 1. right)
What does "full" mean in the case of LaserMix? Since it is a semi-supervised approach, what does training with the full train split mean? Is it just the Cylinder3D/FIDNet trained in a supervised way? If so, why the results don't match? Or is there any mixing involved  like labeled + labeled?

**Comment by ldkong1205** [comment]
> In your paper, you present a comparison with Full Labels (Fig 1. right)
> 
> What does "full" mean in the case of LaserMix? Since it is a semi-supervised approach, what does training with the full train split mean? Is it just the Cylinder3D/FIDNet trained in a supervised way? If so, why the results don't match? Or is there any mixing involved  like labeled + labeled?

Hi @matildecc, thanks for your interest in our work.

For your question: The "full" setting means that the whole training set is considered as both the labeled set and unlabeled set in semi-supervised learning. The framework is LaserMix, which takes a pair of labeled and unlabeled data for feature learning. Therefore, the performance is higher than simply using Cylinder3D/FIDNet for training on the labeled set.

Should you have any questions, please let me know. Thanks again!

**Comment by matildecc** [comment]
Hi @ldkong1205 !
Thank you for your quick answer. And congrats for the work. It's really interesting...

I don't think I understood your answer correctly. 
When it's 10%, it means that 10% of the original train split is selected as labeled and 90% as unlabeled. The same idea applies to 20% and 50%. If you're referring to the "full" case, it would mean that 100% of the data is labeled? If yes, how do you perform the mixing part? labeled + labeled? Do the pseudo-labeling still exists ?

**Comment by ldkong1205** [comment]
> Hi @ldkong1205 ! Thank you for your quick answer. And congrats for the work. It's really interesting...
> 
> I don't think I understood your answer correctly. When it's 10%, it means that 10% of the original train split is selected as labeled and 90% as unlabeled. The same idea applies to 20% and 50%. If you're referring to the "full" case, it would mean that 100% of the data is labeled? If yes, how do you perform the mixing part? labeled + labeled? Do the pseudo-labeling still exists ?

Hi @matildecc, sorry for the late reply!

For your question:
- *"When it's 10%, it means that 10% of the original train split is selected as labeled and 90% as unlabeled."* 
  - **A:** Your understanding is correct. This is the standard setting for semi-supervised learning.
- *If you're referring to the "full" case, it would mean that 100% of the data is labeled? If yes, how do you perform the mixing part? labeled + labeled? Does the pseudo-labeling still exist?* 
  - **A:** In terms of 100%, we assume that all training samples are both treated as "labeled" and "unlabeled". We perform the mixing just as semi-supervised learning. The pseudo-labels for the "unlabeled" (although they have labels under 100% setting) are still generated from the Teacher net. You can think of this as some sort of "data augmentation" of "self-training", where the training samples are supervised by both the ground truth labels, as well as the pseudo-labels generated by the network itself.

Please let us know if the above resolves your question. We are happy to provide more detailed explanations if you still have issues!
